diff --git a/cmake/ConfigureCompiler.cmake b/cmake/ConfigureCompiler.cmake
index 3a83628..9ee8b02 100644
--- a/cmake/ConfigureCompiler.cmake
+++ b/cmake/ConfigureCompiler.cmake
@@ -155,6 +155,10 @@ if(APPLE)
 add_definitions(-D_LIBCPP_ENABLE_CXX17_REMOVED_UNARY_BINARY_FUNCTION)
 endif()
 
+# Fix for std::unary_function being removed in C++17
+# This tells boost to use its alternative implementation
+add_definitions(-DBOOST_NO_CXX98_FUNCTION_BASE)
+
 if (USE_CCACHE)
   find_program(CCACHE_PROGRAM "ccache" REQUIRED)
   set(CMAKE_C_COMPILER_LAUNCHER "${CCACHE_PROGRAM}")
